File: aesi-eclipse/bundles/aesi-eclipse/src/main/java/com/virtlink/aesi/ICancellationToken.java
package com.virtlink.aesi;

/**
 * A cancellation token.
 */
public interface ICancellationToken {

    /**
     * Gets whether the operation has been cancelled.
     * @return {@code true} when the operation has been cancelled;
     * otherwise, {@code false}.
     */
    boolean isCancelled();

}

File: aesi/src/main/kotlin/com/virtlink/editorservices/ICancellationToken.kt
package com.virtlink.editorservices

/**
 * Object that indicates whether the operation has been cancelled.
 *
 * Cancelled operations still need to return, and they may return
 * with either the complete result, an incomplete result, or an error
 * (which may be a cancellation error indicating that the operation
 * has been cancelled and that there is no result).
 */
interface ICancellationToken {

    /**
     * Gets whether the operation has been cancelled.
     */
    val isCancelled: Boolean

}